include/swift/SIL/SILInstruction.h

```diff
@@ -2133,6 +2138,32 @@ class ApplyInstBase<Impl, Base, true>
   bool hasSemantics(StringRef semanticsString) const {
     return doesApplyCalleeHaveSemantics(getCallee(), semanticsString);
   }
+
+// SWIFT_ENABLE_TENSORFLOW
+private:
+  /// Predicate used to filter InoutArgumentRange.
+  struct OperandToInoutArgument {
+    const Impl &inst;
+    OperandToInoutArgument(const Impl &inst) : inst(inst) {}
+    Optional<SILValue> operator()(unsigned long i) const {
+      auto paramInfo = inst.getSubstCalleeConv().getParameters()[i];
+      auto argument = inst.getArgumentsWithoutIndirectResults()[i];
+      if (paramInfo.isIndirectMutating())
```

Comment:
Note: `@inout`-ness is represented in `SILParameterInfo`. It's quite more complicated to get `SILParameterInfo` from just an argument `Operand` (need to use operand index to compute argument index) so I decided to iterate over argument indices.